packages/swayze/lib/src/widgets/headers/gestures/resize_header/header_edge_mouse_listener.dart

```diff
@@ -0,0 +1,278 @@
+  );
+
+  bool _showResizeCursor = false;
+
+  @override
+  void initState() {
+    super.initState();
+
+    resizeNotifier.addListener(_didHoverHeaderEdge);
+    _didHoverHeaderEdge();
+  }
+
+  @override
+  void dispose() {
+    resizeNotifier.removeListener(_didHoverHeaderEdge);
+
+    super.dispose();
+  }
+
+  /// Update the resize cursor if the user is hovering an header edge.
+  void _didHoverHeaderEdge() {
+    final showResizeCursor = resizeNotifier.value != null;
+
+    if (showResizeCursor == _showResizeCursor) {
+      return;
+    }
+
+    setState(() {
+      _showResizeCursor = showResizeCursor;
+    });
+  }
+
+  MouseCursor _getMouseCursor() {
+    if (!_showResizeCursor) {
+      return MouseCursor.defer;
+    }
+
+    return resizeNotifier.value!.axis == Axis.horizontal
+        ? SystemMouseCursors.resizeColumn
+        : SystemMouseCursors.resizeRow;
+  }
+
+  /// Gets the pixel offset for the given [offset] and [axis].
+  double _getOffsetPositionForAxis(Offset offset, Axis axis) {
+    return axis == Axis.horizontal ? offset.dx : offset.dy;
+  }
+
+  /// Checks if the mouse coordinates are at an header edge.
+  void _handleOnHover(PointerHoverEvent event) {
+    Axis? axis;
+
+    // vertical header is being hovered
+    if (event.localPosition.dy > kColumnHeaderHeight &&
+        event.localPosition.dx < kRowHeaderWidth) {
+      axis = Axis.vertical;
+    }
+
+    // horizontal header is being hovered
+    if (event.localPosition.dy < kColumnHeaderHeight) {
+      axis = Axis.horizontal;
+    }
+
+    if (axis != null) {
+      final result = _updateHeaderEdgeDetails(
+        localPosition: event.localPosition,
+        axis: axis,
+      );
+
+      if (result) {
+        return;
+      }
+    }
+
+    resizeNotifier.value = null;
+  }
+
+  /// Updates [resizeNotifier.value] if the user is hovering an headers edge.
+  ///
+  /// Returns `true` if [resizeNotifier.value] has been updated and `false`
+  /// otherwise.
+  bool _updateHeaderEdgeDetails({
+    required Offset localPosition,
+    required Axis axis,
+  }) {
+    var localPixelOffset = _getOffsetPositionForAxis(localPosition, axis);
+
+    final axisContext = viewportContext.getAxisContextFor(axis: axis);
+
+    // since this widget will be placed at a table, we need to take into account
+    // that from `0` to `kRowHeaderWidth` or `kColumnHeaderHeight` (depending on
+    // the axis), there's an empty space. Since the `localPosition` will still
+    // count that into it's offset, we subtract that extent.
+    if (axis == Axis.horizontal) {
+      localPixelOffset -= kRowHeaderWidth;
+    } else {
+      localPixelOffset -= kColumnHeaderHeight;
+    }
+
+    final frozenExtent = axisContext.value.frozenExtent;
+    final hasFrozenHeaders = frozenExtent > 0;
+    final displacement = axisContext.virtualizationState.displacement;
+
+    final ignoreDisplacement = hasFrozenHeaders &&
+        displacement < 0 &&
+        localPixelOffset < frozenExtent + 2;
```

Comment:
Why `+2`? Should we move that to a constant/configurable value and document it?

Reply:
because from `-2` to `+2` from the header separator position still counts as hover. I will move to constant